numpy/_core/src/multiarray/ctors.c

```diff
@@ -2250,6 +2250,12 @@ PyArray_FromInterface(PyObject *origin)
     /* Get dimensions from shape tuple */
     else {
         n = PyTuple_GET_SIZE(attr);
+        if (n > NPY_MAXDIMS) {
+            PyErr_Format( PyExc_ValueError,
+                          "number of dimensions must be within [0, %d], got %d",
+                          NPY_MAXDIMS, n );
```

Comment:
```suggestion
            PyErr_Format(PyExc_ValueError,
                         "number of dimensions must be within [0, %d], got %d",
                         NPY_MAXDIMS, n);
```
Just a nitpick but we never put spaces there.

There is actually another small bug (hard to test here).  `n` should be a `Py_ssize_t`.  I am honestly always surprised that we don't have compiler warnings for an implicit unsafe cast here...